rust/Earthfile

```diff
@@ -1,24 +1,45 @@
 VERSION --global-cache 0.7
 
+# INIT stores the configuration required for the other UDCs in the filesystem, and installs required dependencies.
+#   - cache_id: Overrides default ID of the global $CARGO_HOME cache. Its value is exported to the build environment under the entry: $CARGO_HOME_CACHE_ID
+#   - keep_fingerprints (false): Instructs the following +CARGO calls to don't remove the Cargo fingerprints of the source packages. Use only when source packages have been COPYed with --keep-ts option.
+#   - sweep_days (4): +CARGO calls use cargo-sweep to clean build artifacts that haven't been accessed for this number of days.
+INIT:
+    COMMAND
+    DO +INSTALL_CARGO_SWEEP
+    RUN mkdir -p /earthly/cfg
+
+    # cache_id
+    ARG EARTHLY_TARGET_PROJECT_NO_TAG
+    ARG OS_RELEASE=$(md5sum /etc/os-release | cut -d ' ' -f 1)
+    ARG cache_id="${EARTHLY_TARGET_PROJECT_NO_TAG}#${OS_RELEASE}#earthly-cargo-cache"
+    RUN echo "$cache_id">/earthly/cfg/cache_id
+    ENV CARGO_HOME_CACHE_ID=$cache_id
```

Comment:
This is nice as a workaround but can I suggest you consider making this a first class feature of Earthly UDCs. e.g.

```
IMPORT github.com/earthly/lib/rust AS rust-udc WITH --foo=bar --cache-id=foo
```
To set some `ARG --global` declared at the top of the UDC?

Reply:
This is a good feature to add for sure Ian, mind creating a ticket for it? 
https://github.com/earthly/earthly/issues/new?assignees=&labels=type%3Aproposal&projects=&template=feature-proposal.md&title=

Reply:
https://github.com/earthly/earthly/issues/3488